exporter/sentryexporter/sentry_client.go

```diff
@@ -0,0 +1,314 @@
+	GetAllProjects(ctx context.Context, orgSlug string) ([]projectInfo, error)
+	GetProjectKeys(ctx context.Context, orgSlug, projectSlug string) ([]projectKey, error)
+	GetOrgProjectKeys(ctx context.Context, orgSlug string) ([]projectKey, error)
+	GetOTLPEndpoints(ctx context.Context, orgSlug, projectSlug string) (*otlpEndpoints, error)
+	CreateProject(ctx context.Context, orgSlug, teamSlug, projectSlug, projectName, platform string) (*projectInfo, error)
+}
+
+// sentryClient handles communication with the Sentry API.
+type sentryClient struct {
+	baseURL   string
+	authToken string
+	client    *http.Client
+}
+
+// newSentryClient is used to override the sentry client factory. While running tests we need to mock the http transport, so that
+// we don't open real sockets.
+var newSentryClient = func(baseURL, authToken string, httpClient *http.Client) sentryAPIClient {
+	return newSentryClientImpl(baseURL, authToken, httpClient)
+}
+
+// projectKey represents a Sentry project key.
+type projectKey struct {
+	ID        string   `json:"id"`
+	Name      string   `json:"name"`
+	Public    string   `json:"public"`
+	Secret    string   `json:"secret"`
+	ProjectID int      `json:"projectId"`
+	IsActive  bool     `json:"isActive"`
+	DSN       dsnField `json:"dsn"`
+}
+
+// dsnField represents the DSN field from API response.
+type dsnField struct {
+	Public string `json:"public"`
+}
+
+// ParsePublicDSN parses the public DSN string into a sentry.Dsn.
+func (d *dsnField) ParsePublicDSN() (*sentry.Dsn, error) {
+	return sentry.NewDsn(d.Public)
+}
+
+// teamInfo represents a Sentry team.
+type teamInfo struct {
+	ID   string `json:"id"`
+	Slug string `json:"slug"`
+	Name string `json:"name"`
+}
+
+// projectInfo represents a Sentry project.
+type projectInfo struct {
+	ID    string     `json:"id"`
+	Slug  string     `json:"slug"`
+	Name  string     `json:"name"`
+	Team  teamInfo   `json:"team"`
+	Teams []teamInfo `json:"teams"`
+}
+
+// otlpEndpoints contains the OTLP endpoint URLs for a project.
+type otlpEndpoints struct {
+	TracesURL string
+	LogsURL   string
+	PublicKey string
+}
+
+// sentryAPIError represents an error from Sentry Management API.
+type sentryAPIError struct {
+	StatusCode int
+	Body       string
+	Headers    http.Header
+}
+
+func (e *sentryAPIError) Error() string {
+	return fmt.Sprintf("request failed with status %d: %s", e.StatusCode, e.Body)
+}
+
+func parseProjectID(id string) int {
+	projectID, err := strconv.Atoi(id)
+	if err != nil {
+		return -1
+	}
+	return projectID
+}
+
+func newSentryClientImpl(baseURL, authToken string, httpClient *http.Client) *sentryClient {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+	return &sentryClient{
+		baseURL:   strings.TrimSuffix(baseURL, "/"),
+		authToken: authToken,
+		client:    httpClient,
+	}
+}
+
+// GetAllProjects fetches all projects for a given organization.
+func (c *sentryClient) GetAllProjects(ctx context.Context, orgSlug string) ([]projectInfo, error) {
```

Comment:
m: There's quite a lot of duplication in all of these API methods. Can we extract them into a common helper?